### PR TITLE
[`bug`] support for large `forward_batch_size` in seq2seq models

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

First of all, please welcome the PR number 100! 

Before this PR, it was not possible to run the T5 example with `forward_batch_size > 1`. We suspected that something was wrong with the computation of the loss function when padding tokens are present.

This PR fixes a bug we had for seq2seq models (and I believe for causalLM models too). It seems that we forgot to mask out the logits/hidden states that correspond to the pad tokens when computing the loss function. 

See a similar implementation here: https://github.com/CarperAI/trlx/blob/main/trlx/trainer/nn/ppo_models.py#L166-L191 where the loss takes care of ignoring the terms corresponding to pad tokens

Can also add tests! 

GPT2 run: https://wandb.ai/distill-bloom/trl/runs/4cs5z6j3?workspace=user-younesbelkada
T5 run: https://wandb.ai/distill-bloom/trl/runs/lxgi5ae9?workspace=user-younesbelkada 
It seems that now the `reward_std` is higher, leading to less smooth `reward_mean` curves, so putting this PR as draft

cc @lvwerra

### CODE REVIEW

**Comment by lvwerra** [comment]
Closing in favour of #133.
